schematic/schemas/json_schema_generator.py

```diff
@@ -0,0 +1,747 @@
+        schema[name]["type"] = property_data.property_type
+    elif property_data.property_type:
+        schema[name]["oneOf"] = [
+            {"type": property_data.property_type, "title": property_data.property_type},
+            {"type": "null", "title": "null"},
+        ]
+    elif is_required:
+        schema[name]["not"] = {"type": "null"}
+
+    if property_data.range_min is not None:
+        schema[name]["minimum"] = property_data.range_min
+    if property_data.range_max is not None:
+        schema[name]["maximum"] = property_data.range_max
+
+    return schema
+
+
+def _get_property_data_from_validation_rules(
+    validation_rules: list[str],
+) -> PropertyData:
+    """
+    Returns data from validation rules in the form of a PropertyData dataclass
+
+    Args:
+        validation_rules: A list of validation rules
+
+    Returns:
+        PropertyData: validation rule data
+    """
+    property_type: Union[str, None] = None
+    is_array = False
+    range_min: Union[float, None] = None
+    range_max: Union[float, None] = None
+
+    if validation_rules:
+        if rule_in_rule_list("list", validation_rules):
+            is_array = True
+
+        type_rule = _get_type_rule_from_rule_list(validation_rules)
+        if type_rule:
+            property_type = TYPE_RULES.get(type_rule)
+
+        range_rule = _get_in_range_rule_from_rule_list(validation_rules)
+        if range_rule:
+            if property_type not in ["number", "integer"]:
+                property_type = "number"
+            range_min, range_max = _get_ranges_from_range_rule(range_rule)
+
+    return PropertyData(property_type, is_array, range_min, range_max)
+
+
+def _get_ranges_from_range_rule(
+    rule: str,
+) -> tuple[Union[float, None], Union[float, None]]:
+    """
+    Returns the min and max from an inRange rule if they exist
+
+    Arguments:
+        rule: The inRange rule
+
+    Returns:
+        The min and max form the rule
+    """
+    range_min: Union[float, None] = None
+    range_max: Union[float, None] = None
+    parameters = rule.split(" ")
+    if len(parameters) > 1 and parameters[1].isnumeric():
+        range_min = float(parameters[1])
+    if len(parameters) > 2 and parameters[2].isnumeric():
+        range_max = float(parameters[2])
+    return (range_min, range_max)
+
+
+def _get_in_range_rule_from_rule_list(rule_list: list[str]) -> Union[str, None]:
+    """
+    Returns the inRange rule from a list of rules if there is only one
+    Returns None if there are no inRange rules
+
+    Arguments:
+        rule_list: A list of validation rules
+
+    Raises:
+        ValueError: When more than one inRange rule is found
+
+    Returns:
+        The inRange rule if one is found, or None
+    """
+    in_range_rules = [rule for rule in rule_list if rule.startswith("inRange")]
+    if len(in_range_rules) > 1:
+        raise ValueError("Found more than one inRange rule in validation rules")
```

Comment:
We should probably include the `in_range_rules` that were found so it's more obvious when debugging what to look for.